src/GafferSceneUI/SceneGadget.cpp

```diff
@@ -780,6 +784,51 @@ Imath::Box3f SceneGadget::bound() const
 	return bound( /* selection = */ false );
 }
 
+void SceneGadget::snapshotToFile(
+	const std::filesystem::path &fileName,
+	const Box2f &resolutionGate,
+	const CompoundData *metadata
+) const
+{
+	if( !m_outputBuffer )
+	{
+		return;
+	}
+
+	std::filesystem::create_directories( fileName.parent_path() );
+
+	const Box2i &dataWindow = m_outputBuffer->dataWindow();
+	OIIO::ImageSpec spec( dataWindow.size().x + 1, dataWindow.size().y + 1, 4, OIIO::TypeDesc::FLOAT );
+
+	if( !resolutionGate.isEmpty() )
+	{
+		spec.x = -resolutionGate.min.x;
+		spec.y = -resolutionGate.min.y;
+
+		spec.full_x = 0;
+		spec.full_y = 0;
+		spec.full_width = resolutionGate.size().x;
+		spec.full_height = resolutionGate.size().y;
+	}
+
+	const std::vector<float> &rgbaBuffer = m_outputBuffer->rgbaBuffer();
+
+	const IECoreImage::OpenImageIOAlgo::DataView dataView( metadata );
+	for( const auto &[key, value] : metadata->readable() )
+	{
+		const IECoreImage::OpenImageIOAlgo::DataView dataView( value.get() );
+		if( dataView.data )
+		{
+			spec.attribute( key.c_str(), dataView.type, dataView.data );
+		}
+	}
+
+	std::unique_ptr<OIIO::ImageOutput> output = OIIO::ImageOutput::create( fileName.c_str() );
```

Comment:
I wonder if we should be writing half-float images when the format is EXR? That's what we write when an ImageNode is dragged to the Catalogue. Any opinion @murraystevenson?

Reply:
That sounds good to me, I suppose full float data for the viewport is a little excessive. I'll defer to @murraystevenson for a final go or no go.